### PR TITLE
Customer custom notes

### PR DESCRIPTION
**Pull request by chaotixkilla**

### CODE REVIEW

**Comment by chaotixkilla** [comment]
[Add editable notes section to customer profile page](https://trello.com/c/vWAOREVw/900-add-editable-notes-section-to-customer-profile-page)

File: lib/customers.js [hunk 1]
```diff
@@ -515,6 +518,9 @@ function getCustomerById (id) {
       from cash_in_txs where send_confirmed = true union
       select 'cashOut' as tx_class, id, fiat, fiat_code, created, customer_id 
       from cash_out_txs where confirmed_at is not null) t on c.id = t.customer_id
+      left outer join (
+        select customer_id, content from customer_notes
+      ) cn on c.id = cn.customer_id
```
Comment (josepfo) [inline lib/customers.js]:
```suggestion
      LEFT OUTER JOIN (
        SELECT customer_id, content FROM customer_notes
      ) cn ON c.id = cn.customer_id
```

**Review by josepfo — CHANGES_REQUESTED** [review]

**Review by josepfo — APPROVED** [review]

File: migrations/1627868356883-customer-custom-notes.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+exports.up = function (next) {
+  var sql = [
+    `CREATE TABLE customer_notes (
+      id UUID PRIMARY KEY,
+      customer_id UUID NOT NULL REFERENCES customers(id),
+      created TIMESTAMPTZ NOT NULL DEFAULT now(),
+      last_edited_at TIMESTAMPTZ,
+      last_edited_by UUID REFERENCES users(id),
+      content TEXT
```
Comment (RafaelTaranto) [inline migrations/1627868356883-customer-custom-notes.js]: Any reason for content allowed as null?

File: lib/new-admin/graphql/resolvers/customer.resolver.js [hunk 3]
```diff
@@ -17,6 +17,10 @@ const resolvers = {
       const token = !!context.req.cookies.lid && context.req.session.user.id
       if (customerId === anonymous.uuid) return customers.getCustomerById(customerId)
       return customers.updateCustomer(customerId, customerInput, token)
+    },
+    setCustomerNotes: (...[, { customerId, newContent }, context]) => {
+      const token = !!context.req.cookies.lid && context.req.session.user.id
```
Comment (RafaelTaranto) [inline lib/new-admin/graphql/resolvers/customer.resolver.js]: Can't token be false here? And in other places we're doing the same? I would rather have a get token function that returns null and the null is properly handled. I know this probably won't happen but I think it's still the better option here.

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**